### PR TITLE
Allow to reset admin user password via CLI

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #15283
| License         | MIT

The feature to log out the user will be added in a separate PR

### CODE REVIEW

File: src/Sylius/Behat/Context/Cli/ChangeAdminPasswordContext.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    /**
+     * @Then I should be able to log in as :email authenticated by :password password
+     */
+    public function iShouldBeAbleToLoginWithEmailAndPassword(string $email = '', string $password = ''): void
+    {
+        /** @var AdminUserInterface|null $adminUser */
+        $adminUser = $this->adminUserRepository->findOneByEmail($email);
+        $adminUser->setPlainPassword($password);
+        Assert::same($adminUser->getPassword(), $this->userPasswordHasher->hash($adminUser));
```
Comment (NoResponseMate) [inline src/Sylius/Behat/Context/Cli/ChangeAdminPasswordContext.php]:
```suggestion
        $adminUser->setPlainPassword($password);

        Assert::same($adminUser->getPassword(), $this->userPasswordHasher->hash($adminUser));
```

File: src/Sylius/Behat/Resources/config/suites.yml [hunk 2]
```diff
@@ -71,6 +71,7 @@ imports:
 
     - suites/cli/canceling_unpaid_orders.yml
     - suites/cli/installer.yml
+    - suites/cli/change_admin_password.yml
```
Comment (NoResponseMate) [inline src/Sylius/Behat/Resources/config/suites.yml]:
```suggestion
    - suites/cli/change_admin_password.yml
    - suites/cli/installer.yml
```

File: src/Sylius/Bundle/AdminBundle/Command/Factory/QuestionFactory.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+use Symfony\Component\Console\Question\Question;
+
+final class QuestionFactory implements QuestionFactoryInterface
+{
+    public function createEmail(): Question
+    {
+        $question = new Question('Email');
+        $question->setValidator(function (?string $email) {
+            if (!filter_var($email, \FILTER_VALIDATE_EMAIL) || $email === null) {
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Command/Factory/QuestionFactory.php]:
```suggestion
            if ($email === null || !filter_var($email, \FILTER_VALIDATE_EMAIL)) {
```

File: src/Sylius/Bundle/AdminBundle/Tests/Command/ChangeAdminUserPasswordCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->userRepository = $this->createMock(UserRepositoryInterface::class);
+
+        $this->passwordUpdater = $this->createMock(PasswordUpdaterInterface::class);
+
+        $this->questionFactory = $this->createMock(QuestionFactoryInterface::class);
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Tests/Command/ChangeAdminUserPasswordCommandTest.php]:
```suggestion
        $this->userRepository = $this->createMock(UserRepositoryInterface::class);
        $this->passwordUpdater = $this->createMock(PasswordUpdaterInterface::class);
        $this->questionFactory = $this->createMock(QuestionFactoryInterface::class);
```

**Review by NoResponseMate — DISMISSED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Review by TheMilek — APPROVED** [review]

**Review by jakubtobiasz — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @Wojdylak!

**Pull request merged**